src/api/general-services/pipeline-manage/constructors/assign-pod-to-pipeline.js

```diff
@@ -132,40 +149,105 @@ const assignPodToPipeline = (context, step) => {
               Next: 'NoPodsAvailable',
             },
           ],
-          Default: 'AssignPodToPipeline',
+          Default: 'RequestPod',
         },
         NoPodsAvailable: {
           Type: 'Fail',
           Cause: 'No available and running pipeline pods.',
           Error: 'NoPodsAvailable',
         },
-        AssignPodToPipeline: {
+        RequestPod: {
+          Comment: 'Send a message through SNS so that the API assigns a pod to the pipeline',
+          Type: 'Task',
+          Resource: 'arn:aws:states:::sns:publish',
           End: true,
-          ResultPath: '$.PatchResult',
+          Parameters: {
+            // eslint-disable-next-line max-len
+            TopicArn: `arn:aws:sns:${config.awsRegion}:${accountId}:work-results-${environment}-${sandboxId}`,
+            Message: JSON.stringify(requestPodMessage),
+            MessageAttributes: {
+              type: {
+                DataType: 'String',
+                StringValue: 'PipelineResponse',
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+};
+
+
+const waitForPod = (context, step) => {
+  const { clusterInfo, activityArn } = context;
+
+  const activityId = getActivityId(activityArn);
+
+  return {
+    ...step,
+    Type: 'Map',
+    ItemsPath: '$.retries',
+    MaxConcurrency: 1,
+    // retry waits up to 74 seconds for the pod to be assigned
```

Comment:
It may be worth adding a formula for arbitrary values in case we need to change it. The formula is:

```
t = I(1 - b^(m+1)/(1-b))
```

Derivation:

For an `IntervalSeconds` of `I`, a `MaxAttempts` of `m` and a `BackoffRate` of `b`, the total wait time is:

![image](https://user-images.githubusercontent.com/16019444/136945919-9270f7ac-4c2d-4f61-a5ee-4b28b1eda00a.png)

Factoring out `I` we get:

![image](https://user-images.githubusercontent.com/16019444/136945992-bfebc599-8d22-4c1c-9184-c6ce34364ee8.png)

The total wait time `t` is just `I` times the sum, which has a closed form solution:

```
S(m)   = b^0 + b^1 + b^2 + ... + b^m
b*S(m) =       b^1 + b^2 + ... + b^m + b^(m+1)

S(m) - b*S(m) = 1 - b^(m+1)
S(m) * (1-b)  = 1 - b^(m+1)
S(m)          = 1 - b^(m+1)/(1-b)
```